src/core/manager.c

```diff
@@ -2748,6 +2801,142 @@ static int manager_dispatch_notify_fd(sd_event_source *source, int fd, uint32_t
         return 0;
 }
 
+#if HAVE_PAM
+static int manager_dispatch_notify_pam_pid_fd(sd_event_source *source, int fd, uint32_t revents, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
+        char buf[DECIMAL_STR_MAX(pid_t) * 2 + STRLEN("\n") + 1];
+        struct iovec iovec = {
+                .iov_base = buf,
+                .iov_len = sizeof(buf)-1,
+        };
+        CMSG_BUFFER_TYPE(CMSG_SPACE(sizeof(struct ucred)) + CMSG_SPACE(sizeof(int))) control;
+        struct msghdr msghdr = {
+                .msg_iov = &iovec,
+                .msg_iovlen = 1,
+                .msg_control = &control,
+                .msg_controllen = sizeof(control),
+        };
+        ssize_t n;
+        int r;
+
+        assert(m->notify_pam_pid_fds[0] == fd);
+
+        if (revents != EPOLLIN) {
+                log_warning("Got unexpected poll event for notify fd.");
+                return 0;
```

Comment:
It seems that it'd be cleaner to return normally from this function and suppress the return value in the caller. 
```C
return log_warning_errno(SYNTHETHIC_ERRNO(EAGAIN), …);
```